core/contracts/Voting.sol

```diff
@@ -27,61 +27,55 @@ contract Voting is Testable, MultiRole, OracleInterface {
     using ResultComputation for ResultComputation.Data;
 
     // Identifies a unique price request for which the Oracle will always return the same value.
+    // Tracks ongoing votes as well as the result of the vote.
     struct PriceRequest {
         bytes32 identifier;
         uint time;
-    }
 
-    struct Round {
-        // The list of price requests that were/are being voted on this round.
-        PriceRequest[] priceRequests;
+        mapping(uint => VoteRound) voteRounds;
 
-        // Voting token snapshot ID for this round. If this is 0, no snapshot has been taken.
-        uint snapshotId;
-
-        // Inflation rate set for this round.
-        FixedPoint.Unsigned inflationRate;
+        // The final result of a vote.
+        // If lastVotingRound < voteTiming.getLastUpdatedRoundId, vote was resolved.
+        int resolvedPrice;
+        uint lastVotingRound;
     }
 
-    // A particular voter's submission.
-    struct VoteSubmission {
-        // A bytes32 of `0` indicates no commit or a commit that was already revealed.
-        bytes32 committedHash;
+    struct VoteRound {
+        mapping(address => Vote) votes;
 
-        // The value of the vote that was revealed.
-        int revealedVote;
-
-        // Whether the voter revealed their vote (this is to handle valid reveals for 0 prices).
-        bool didReveal;
+        ResultComputation.Data resultComputation;
     }
 
-    struct VoteInstance {
-        // Maps (voterAddress) to their submission.
-        mapping(address => VoteSubmission) voteSubmissions;
-
-        // The data structure containing the computed voting results.
-        ResultComputation.Data resultComputation;
+    struct Vote {
```

Comment:
`VoteInstance` -> `VoteRound`
`VoteSubmission` -> `Vote`

What do you think?

Reply:
Does this get confusing since `Round` would contain many `VoteRound`s?

Reply:
Maybe `Ballot` and `Vote`? Or `Vote` and `Election`? Or `Poll` and `Vote`?

Reply:
* I agree that `VoteRound` conflicts with `Round`. 
* `Ballot`, `Election`, and `Poll` all seem to have too much connotation.

I think the problem here is that we're indexing by `roundId` twice. Once in `mapping(uint => Round) private rounds` and again in `mapping(uint => VoteInstance) voteInstance`. I'll revert this section and maybe iterate in a separate PR.